docs/csharp/language-reference/keywords/codesnippet/CSharp/event_1.cs

```diff
@@ -14,10 +14,12 @@ public class Publisher
 
         // Wrap the event in a protected virtual method
         // to enable derived classes to raise the event.
-        protected virtual void RaiseSampleEvent()
-        {
-            // Raise the event by using the () operator.
-            if (SampleEvent != null)
-                SampleEvent(this, new SampleEventArgs("Hello"));
-        }
+    protected virtual void OnRaiseSampleEvent()
+    {
+        //Create a temporary reference to guard against a race condition
```

Comment:
Thanks for catching this.

We've been recommending the `?.` operator for this idiom in modern C#. Can you change the code sample to:

```csharp
SampleEvent?.Invoke(this, new SampleEventArgs("Hello"));
```

It is semantically the same, and cleaner.